src/Processors/Formats/Impl/JSONColumnsBlockOutputFormatBase.h

```diff
@@ -12,10 +13,10 @@ class WriteBuffer;
 
 /// Base class for Columnar JSON output formats.
 /// It buffers all data and outputs it as a single block in writeSuffix() method.
-class JSONColumnsBlockOutputFormatBase : public IOutputFormat
+class JSONColumnsBlockOutputFormatBase : public IOutputFormatWithUTF8Validation
```

Comment:
Squeezing the UTF-8 validation into the base class seems not good design. Maybe it could be passed as an argument to a generic base class "IOutputFormat"?

Reply:
To be honest, I don't see why it's not a good design. Using this intermidiate class we can reduce code duplication (previousely, each output format with utf-8 validation had absolutely the same code) and don't clog up the base interface class with features that are used just by some small subset of formats (we already have some specific methods/fields in this base class, so I don't want to make it more and more complex).

Reply:
Thinking about it again, a base class is fine for the chosen adaptor approach.